pkg/cli/debug/debug.go

```diff
@@ -1063,7 +1061,42 @@ func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*
 					},
 				},
 			},
-		}, nil
+		}
+		if t.Labels[corev1.LabelOSStable] == string(corev1.Windows) {
+			template.Spec.OS = &corev1.PodOS{Name: corev1.Windows}
+			containerUser := "ContainerUser"
+			// Override all fields that will throw an error if used on a Windows Pod
+			template.Spec.HostPID = false
+			template.Spec.HostIPC = false
```

Comment:
I don't think it is feasible to explicitly setting all these fields. Because their defaults are already nil or false or empty. What will happen when there is a new field arise?. I'd recommend reverting to the previous version of this PR and QE does verification on it.

Reply:
Makes sense, I misunderstood your previous comment.
I will ask QE to do testing against the new revision

Reply:
ah, sorry, I should have been more clear.